src/libtorch.cc

```diff
@@ -59,6 +59,107 @@
 
 namespace triton { namespace backend { namespace pytorch {
 
+namespace {
+
+// This object is used to capture timestamps for the instance group of type
+// 'MODEL'.
+class TimestampData {
+ public:
+  TimestampData()
+      : compute_input_start_(0), earliest_compute_infer_start_(0),
+        last_compute_infer_start_(0), compute_output_start_(0)
+  {
+  }
+
+  enum struct TimestampHint { COMPUTE_INPUT, COMPUTE_INFER, COMPUTE_OUTPUT };
+
+  // Set the timestamp hint
+  void SetTimestampHint(const TimestampHint& hint) { hint_ = hint; }
```

Comment:
Instead of setting the hint separately how about making it an argument to `CaptureTimestamp`?

Reply:
The `CaptureTimestamp` function is called inside the callback function where only one argument can be passed: https://github.com/triton-inference-server/pytorch_backend/pull/107/files#diff-859dcc1d94f2d13b0d19e7a14af430cd18eb66395a6f69f2d4886854319c4467R154. We could pass both `timestamp_data_` and `hint` as a tuple to the callback but I thought using a separate function makes the code more straightforward.

Reply:
How about having two variants of `TimestampCaptureCallback` function in this class? One with timestamp override to capture the last timestamp and the other without timestamp override to capture the first timestamp?

Reply:
Used two callback function to capture the first the the last timestamp.